django/db/models/base.py

```diff
@@ -1838,8 +1838,13 @@ def _check_constraints(cls):
             if not router.allow_migrate_model(db, cls):
                 continue
             connection = connections[db]
+            try:
+                supports_table_check_constraints = connection.features.supports_table_check_constraints
+            except DatabaseError:
+                # Could not establish a connection, disable check instead of crashing.
+                continue
             if (
-                connection.features.supports_table_check_constraints or
+                supports_table_check_constraints or
```

Comment:
I'm afraid that we can hit this in other cases, e.g. in [_check_json_fields()](https://github.com/django/django/pull/11452/files#diff-507b415116b409afa4f723e41a759a9eR1340-R1360) in #11452. What do you think about checking connection usability in `mysql_server_info()`, e.g.
```python
    @cached_property
    def mysql_server_info(self):
        if not self.is_usable():
            return ''
        with self.temporary_connection() as cursor:
            cursor.execute('SELECT VERSION()')
            return cursor.fetchone()[0]
```

Reply:
@felixxm that's what I initially tried but it was crashing with `AttributeError`

```
  File "/tests/django/django/db/backends/mysql/base.py", line 351, in mysql_server_info
    if not self.is_usable():
  File "/tests/django/django/db/backends/mysql/base.py", line 329, in is_usable
    self.connection.ping()
AttributeError: 'NoneType' object has no attribute 'ping'
```

Reply:
So maybe we should check connection in `is_usable()` :thinking: like in related fix for PostgreSQL
https://github.com/django/django/commit/a96b9019320ed8236659ee520a7a017c1bafbc6f#diff-aef216ccdc37941bfed98ea3c80f7642R251-R252